content/200-concepts/100-components/02-prisma-client/260-client-extensions/130-result.mdx

```diff
@@ -0,0 +1,152 @@
+
+```ts
+const xprisma = prisma.$extends({
+  name?: 'name',
+  result?: {
+    user: {                   // in this case, we extend the `user` model
+      myComputedField: {      // the name of the new computed field
+        needs: { ... }
+        compute(): { ... }
+      },
+    },
+  },
+});
+```
+
+The parameters are as follows:
+
+- `name`: (optional) specifies a name for the extension that appears in error logs.
+- `result`: defines new fields and methods to the query results.
+- `needs`: an object which describes the dependencies of the result field.
+- `compute`: an object that contains methods that define how the virtual field is computed when it is accessed. You must declare the same properities in both `compute` and `needs`.
+
+### Add a custom field to query results
+
+You can use the `result` extension component to add fields to query results. These fields are computed at runtime and are type-safe.
+
+In the following example, we add a new virtual field called `fullName` to the `user` model.
+
+```ts
+const xprisma = prisma.$extends({
+  result: {
+    user: {
+      fullName: {
+        needs: { firstName: true, lastName: true }, // the dependencies
+        compute(user) {
+          // the computation logic
+          return `${user.firstName} ${user.lastName}`
+        },
+      },
+    },
+  },
+})
+
+const user = await xprisma.user.findFirst()
+
+// return the user's full name, such as "John Doe"
+console.log(user.fullName)
+```
+
+In above example, the input `user` is automatically typed according to the object defined in `needs`. `firstName` and `lastName` are of type `string`, because they are specified in `needs`. If they are not specified in `needs`, then their type is `string | undefined`.
+
+When you call a method in an extension, use the constant name from your `$extends` statement, not `prisma`. In the above example, `xprisma.user.findFirst` works, but `prisma.user.findFirst` does not.
+
+### Add a computed field to query results
+
+The following example computes a user's title and full name in a type-safe way.
+
+```ts
+const xprisma = prisma
+  .$extends({
+    result: {
+      user: {
+        fullName: {
+          needs: { firstName: true, lastName: true },
+          compute(user) {
+            return `${user.firstName} ${user.lastName}`
+          },
+        },
+      },
+    },
+  })
+  .$extends({
+    result: {
+      user: {
+        titleFullName: {
+          needs: { title: true, fullName: true },
+          compute(user) {
+            return `${user.title} (${user.fullName})`
+          },
+        },
+      },
+    },
+  })
+```
+
+#### Considerations for fields
+
+- For performance reasons, Prisma Client computes results when your application accesses them at runtime, not on retrieval.
+- You can only create fields that are based on scalar fields.
+- You can only use computed fields with `select` and you cannot aggregate them.
+
+  ```ts
```

Comment:
Is this for the statement just above?

Reply:
Yes it is. I'll put in a "for example" to make that clearer.